nmtwizard/preprocess/loader.py

```diff
@@ -100,22 +100,31 @@ def close_files(self):
 
     def __call__(self):
 
-        tu_batch = []
+        tu_list = []
         while True:
-            del tu_batch[:] # TODO V2: Check memory usage on a big corpus
+            del tu_list[:] # TODO V2: Check memory usage on a big corpus
             # Read sampled lines from all files and build TUs.
             batch_line = 0
+            annotations = self._file.files.get("annotations", {})
             while (batch_line < self._batch_size and self._current_line < self._file.lines_count):
-                src_line = self._file.files[0].readline().strip()
-                tgt_line = self._file.files[1].readline().strip()
+                src_line = self._file.files["src"].readline().strip()
+                tgt_line = self._file.files["tgt"].readline().strip()
+                annot_lines = {}
+                for key, annot_file in annotations.items():
+                    annot_lines[key] = annot_file.readline().strip()
                 if (self._current_line in self._file.random_sample):
                     while self._file.random_sample[self._current_line] and \
                           batch_line < self._batch_size:
-                        tu_batch.append(tu.TranslationUnit((src_line, tgt_line)))
+                        tu_list.append(tu.TranslationUnit((src_line, tgt_line), annotations=annot_lines))
                         batch_line += 1
                         self._file.random_sample[self._current_line] -= 1
                 self._current_line += 1
-            if not tu_batch:
+            if not tu_list:
                 return
 
-            yield tu_batch
+            meta = self._file.weight.copy()
+            meta["base_name"] = self._file.base_name
+            meta["root"] = self._file.root
+            meta["no_preprocess"] = self._file.no_preprocess
+
+            yield tu_list, meta
```

Comment:
What is the difference between this meta and the one included in the `TranslationUnit` class?

Reply:
Meta here is batch-level, for the moment it mostly gives the information in training about the file the batch comes from.
Meta in translation unit is about the particular example. It contains context information for example.
Maybe I could find more sensible names for both...